pkg/rungroup/rungroup.go

```diff
@@ -0,0 +1,85 @@
+package rungroup
+
+// rungroup expands on oklog/run, adding logs to indicate which actor caused
+// the interrupt and which actor, if any, is preventing shutdown. In the
+// future, we would like to add the ability to force shutdown before a given
+// timeout. See: https://github.com/kolide/launcher/issues/1205
+
+import (
+	"fmt"
+
+	"github.com/go-kit/kit/log"
+	"github.com/go-kit/kit/log/level"
+)
+
+type (
+	Group struct {
+		logger log.Logger
+		actors []rungroupActor
+	}
+
+	rungroupActor struct {
+		name      string // human-readable identifier for the actor
+		execute   func() error
+		interrupt func(error)
+	}
+
+	actorError struct {
+		errorSourceName string
+		err             error
+	}
+)
+
+func NewRunGroup(logger log.Logger) *Group {
+	return &Group{
+		logger: log.With(logger, "component", "run_group"),
+		actors: make([]rungroupActor, 0),
+	}
+}
+
+func (g *Group) Add(name string, execute func() error, interrupt func(error)) {
+	g.actors = append(g.actors, rungroupActor{name, execute, interrupt})
+}
+
+func (g *Group) Run() error {
+	if len(g.actors) == 0 {
+		return nil
+	}
+
+	// Run each actor.
+	level.Debug(g.logger).Log("msg", "starting all actors", "actor_count", len(g.actors))
+	errors := make(chan actorError, len(g.actors))
+	for _, a := range g.actors {
+		go func(a rungroupActor) {
```

Comment:
This is an unusual usage of an anonymous function.

If you're going to declare one that that takes an `actor` you could make a `func (g *group) start(a rungroupActor)`

or if it's anonymous, just use the `a` defined in the forloop

Reply:
Needing `a := a` is a good point. I dunno, it's not usually what I see in for loops.